src/main/Security/EncryptionController.java

```diff
@@ -0,0 +1,158 @@
+package Security;
+
+import Logger.LogFactory;
+import User.User;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.JsonKeysetReader;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.config.TinkConfig;
+import com.google.crypto.tink.integration.gcpkms.GcpKmsClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.apache.pdfbox.io.IOUtils;
+import org.bson.Document;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.Objects;
+
+import static com.mongodb.client.model.Filters.eq;
+
+public class EncryptionController {
+  Logger logger;
+  MongoDatabase db;
+  Aead aead;
+
+  public static final String masterKeyUri = Objects.requireNonNull(System.getenv("MASTERKEYURI"));
+
+  public static final String credentials =
+      Objects.requireNonNull(System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+
+  public EncryptionController(MongoDatabase db) throws GeneralSecurityException, IOException {
+    this.db = db;
+    LogFactory l = new LogFactory();
+    logger = l.createLogger("EncryptionController");
+    generateAead();
+  }
+
+  // Generates an AEAD Object through Google Tink for encryption and decryption
+  public void generateAead() throws GeneralSecurityException, IOException {
```

Comment:
can we make it not generate an AEAD in the constructor? and then we can leave the function generateAEAD

Reply:
Well, it used to be generateAEAD for each separate method. But this was very inefficient and noticably slow, since it was pulling from the server every time. The AEAD is the same every time, so it makes sense to do it once.